rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp

```diff
@@ -84,62 +74,81 @@ Decoder::Status FusedCropTJDecoder::decode(unsigned char *input_buffer, size_t i
+                double aspect_ratio = std::exp(log_ratio_dist(_rand_gen));
+                crop_width = static_cast<size_t>(std::round(std::sqrt(target_area * aspect_ratio)));
+                crop_height = static_cast<size_t>(std::round(std::sqrt(target_area * (1 / aspect_ratio))));
+                if (is_valid_crop(crop_height, crop_width, original_image_height, original_image_width)) {
+                    x1 = std::uniform_int_distribution<int>(0, original_image_width - crop_width)(_rand_gen);
+                    y1 = std::uniform_int_distribution<int>(0, original_image_height - crop_height)(_rand_gen);
+                    break;
+                }
             }
         }
-        constexpr static float ASPECT_RATIO_RANGE[2] = {0.75, 1.33};
         // Fallback on Central Crop
-        if( !is_valid_crop(crop_height, crop_width, original_image_height, original_image_width)) 
-        {
-            float in_ratio;
-            in_ratio = static_cast<float>(original_image_width) / original_image_height;
-            if(in_ratio < ASPECT_RATIO_RANGE[0])
-            {
+        if(num_attempts_left <= 0) {
+            double in_ratio;
+            float max_area = area_range[1] * original_image_height * original_image_width;
+            in_ratio = static_cast<double>(original_image_width) / original_image_height;
+            if(in_ratio < aspect_ratio_range[0]) {
                 crop_width =  original_image_width;
-                crop_height = crop_width / ASPECT_RATIO_RANGE[0];
+                crop_height = static_cast<size_t>(std::round(crop_width / aspect_ratio_range[0]));
             }
-            else if(in_ratio > ASPECT_RATIO_RANGE[1])
-            {
+            else if(in_ratio > aspect_ratio_range[1]) {
                 crop_height = original_image_height;
-                crop_width  = crop_height  * ASPECT_RATIO_RANGE[1];
-            } 
-            else
-            {
+                crop_width  = static_cast<size_t>(std::round(crop_height * aspect_ratio_range[1]));
+            } else { // Whole Image
                 crop_height = original_image_height;
                 crop_width  = original_image_width;
             }
-            x1 =  (original_image_width - crop_width) / 2;
-            y1 =  (original_image_height - crop_height) / 2;
+            double scale = std::min(1.0f, max_area / (crop_width * crop_height));
+            crop_width = std::max<int>(1, crop_width * std::sqrt(scale));
+            crop_height = std::max<int>(1, crop_height * std::sqrt(scale));
+            x1 = std::uniform_int_distribution<int>(0, original_image_width - crop_width)(_rand_gen);
+            y1 = std::uniform_int_distribution<int>(0, original_image_height - crop_height)(_rand_gen);
         }
     }
-    
-   // std::cout<<"Fused Crop Decoder <x,y, w, h>: " << x1 << " " << y1 << " " << crop_width << " " << crop_height << std::endl;
+    if(crop_width > max_decoded_width)
```

Comment:
please use std::min instead of comparison

Reply:
changed to use std::min